state/indexer/sink/psql/schema.sql

```diff
@@ -1,32 +1,83 @@
-CREATE TYPE block_event_type AS ENUM ('begin_block', 'end_block', '');
-CREATE TABLE block_events (
-    id SERIAL PRIMARY KEY,
-    key VARCHAR NOT NULL,
-    value VARCHAR NOT NULL,
-    height INTEGER NOT NULL,
-    type block_event_type,
-    created_at TIMESTAMPTZ NOT NULL,
-    chain_id VARCHAR NOT NULL,
-    UNIQUE (key, height)
+/*
+  This file defines the database schema for the PostgresQL ("psql") event sink
+  implementation in Tendermint. The operator must create a database and install
+  this schema before using the database to index events.
+ */
+
+-- The blocks table records metadata about each block.
+-- The block record does not include its events or transactions (see tx_results).
```

Comment:
should we provide some way to facilitate running this/versioning this (e.g. an endpoint in the CLI that creates the schema?) 

also is there a way that this is tested against the other code?

Reply:
Re: Provisioning. Sure, that's a good idea. I don't think it necessarily needs to be part of this PR, but it seems like a useful thing for us to have, and I'd be happy to tackle it separately.

Re: Other code. Do you mean against the rest of the indexer plumbing? There is an existing integration-ish test in `state/indexer` that tests this alongside the `kv` and `null` indexers, but I don't know of any beyond that. Do you have a specific idea for what you'd like to see exercised?

Reply:
- tracking the provisioning work separtly seems good, but I think would be good to do. 
- other code: I meant "making sure that the sql file gets run in tests"

Reply:
Agreed. I filed https://github.com/tendermint/tendermint/issues/6878 to track provisioning.

Re: Other code. I believe that case is covered both by the unit tests in this PR and by the integration tests in `state/indexer`, which each apply the schema to a real database and exercise tests against it.

Reply:
perfect!